hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -221,17 +226,27 @@ private void completeExecute() throws AzureBlobFileSystemException {
     LOG.trace("{} REST operation complete", operationType);
   }
 
+  private void updateClientRequestHeader(AbfsHttpOperation httpOperation,
+      TracingContext tracingContext) {
+    tracingContext.generateClientRequestID();
```

Comment:
Generating the clientRequestId in the toString() method of tracingContext is not a good option as it will re-generate it every time the method is called. Also, we need tracingContext to store the id for consistency
Have renamed toString to constructHeader. Format is passed over config and already available to tracingContext object